lib/client.js

```diff
@@ -104,6 +104,21 @@ function messageHandler(client, event) {
   }
 }
 
+// When doing singular operations and we retrieve and error this function will throw that error.
+function throwUnlessBulk(path, result) {
+  var isSingularOperation = typeof path === 'string';
+
+  // CRUFT: When framework starts always appending errors we can remove the extra results.errors check
+  if(isSingularOperation && result.errors && result.errors[path]) {
+    var err = new Error(result.errors[path].message);
```

Comment:
could we abstract out this since it's used here too: https://github.com/zendesk/zendesk_app_framework_sdk/blob/97db3f093a52b7a79be73918916a0b30511de2c0/lib/client.js#L76-78